src/d3a/models/myco_matcher/external_matcher.py

```diff
@@ -1,140 +1,179 @@
 import json
 import logging
+from typing import Dict, List
 
 import d3a.constants
-from d3a.d3a_core.exceptions import InvalidBidOfferPair
+from d3a.d3a_core.exceptions import (
+    InvalidBidOfferPairException, OfferNotFoundException,
+    BidNotFoundException, MycoValidationException)
 from d3a.d3a_core.redis_connections.redis_area_market_communicator import ResettableCommunicator
-from d3a.models.market import validate_authentic_bid_offer_pair
-from d3a.models.market.market_structures import (
-    BidOfferMatch, offer_or_bid_from_json_string)
+from d3a.models.market import Market
+from d3a.models.market.market_structures import BidOfferMatch
 
 from d3a.models.myco_matcher.base_matcher import BaseMatcher
 
 
 class ExternalMatcher(BaseMatcher):
+    """Class responsible for external bids / offers matching."""
     def __init__(self):
         super().__init__()
         self.simulation_id = d3a.constants.COLLABORATION_ID
         self.myco_ext_conn = None
-        self.channel_prefix = f"external-myco/{self.simulation_id}/"
+        self.channel_prefix = f"external-myco/{self.simulation_id}"
+        self.response_channel = f"{self.channel_prefix}/response"
+        self.events_channel = f"{self.response_channel}/events/"
         self._setup_redis_connection()
         self.area_uuid_markets_mapping = {}
-        self.markets_mapping = {}
+        self.markets_mapping = {}  # Dict[market_id: market] mapping
         self.recommendations = []
 
     def _setup_redis_connection(self):
         self.myco_ext_conn = ResettableCommunicator()
         self.myco_ext_conn.sub_to_multiple_channels(
-            {"external-myco/get-simulation-id": self.get_simulation_id,
-             f"{self.channel_prefix}offers-bids/": self.publish_offers_bids,
-             f"{self.channel_prefix}post-recommendations/": self.match_recommendations})
+            {"external-myco/get-simulation-id": self.publish_simulation_id,
+             f"{self.channel_prefix}/offers-bids/": self.publish_offers_bids,
+             f"{self.channel_prefix}/post-recommendations/": self.match_recommendations})
 
     def publish_offers_bids(self, message):
         """Publish open offers and bids.
 
-        published data are of the following format
-        market_offers_bids_list_mapping = {"market_id" : {"bids": [], "offers": [] }, }
+        published data are of the following format:
+            {"bids_offers": {`market_id` : {"bids": [], "offers": [] }, }}
         """
-        # TODO: message can contain filters
-        data = {"event": "offers_bids_response"}
+        response_data = {"event": "offers_bids_response"}
```

Comment:
Here we are using strings to describe the events here, rather than enums as we usually do with market events. Is there a specific reason for this?

Reply:
Nope, an enum is better than having strings into our code, I had to group them since test1, I'll do it in the context of this PR